pom.xml

```diff
@@ -64,6 +64,24 @@
     </properties>
 
     <profiles>
+        <profile>
+            <id>neo4jv4</id>
+            <activation>
+                <jdk>11</jdk>
```

Comment:
Do you plan to build the maven plugin twice, once with a jdk11, and once with a jdk17, and distribute both?
I might be wrong, but I think that these profiles are only useful for building the maven plugin, not for allowing dynamic switch for maven projects using this plugin.

Reply:
The plugin is built once, the profiles are evaluated and activated at execution time. We already used this mechanism for the transition from Neo4j 2.x to 3.x. Once the PR is merged you can give the 2.1.0-SNAPSHOT a try: build a project using JAVA_HOME pointing to JDK 11 and Neo4j v4 will be used, switch to JDK 17 and Neo4j v5 is getting picked up.

Reply:
Great! I never realized that maven profile were evaluated at runtime, even for plugins and dependencies, but it indeed makes sense, since they're declared outside of the `<build>` section.

Reply:
We released milestone for verifying the automatic profile activation, you can check by upgrading to 2.1.0-M1.

Reply:
Thanks! I tested it today, and it works fine! Congrats :-)

Reply:
@DirkMahler : However I don't think this solution will work if I want to develop a jqassistant plugin with java 17+.
Maybe the same strategy (neo4j version dependent of runtime jdk) should be applied in the com.buschmais.jqassistant.core:store and com.buschmais.jqassistant.core:test, so any jqassistant plugin could use the desired java version.
WDYT?

Reply:
The current Neo4j target is 4.x, at least for the default plugins distributed with the distributions compatibility shall be ensured. That's the reason why the test library is still hard-wired to that version of the database. But we could check, if this can be solved another way, i.e. enabling auto-selection of the Neo4j backend for the test artifact as well but deferring the decision to the plugins themselves and forcing JDK 11/Neo4j 4.x using Maven toolchain. If this is needed please create an issue for the core framework and let's refine the approach there.